docs/dev_guides/settings_validation.md

```diff
@@ -0,0 +1,142 @@
+
+Currently, comparison and blocking rule checks are organised in a modular format.
+
+To expand the current suite of tests, you should:
+
+1. Create a function to inspect the presence of the error you're evaluating.
+2. Define an error message that you intend to add to the `ErrorLogger` class.
+3. Integrate these elements into either the `validate_comparison_levels` function or a similar one, which appends any detected errors.
+4. Finally, work out where this function should live in the setup process of the linker object.
+
+
+### Additional Column Checks
+
+Should you need to include extra checks to assess the validity of columns supplied by a user, your primary focus should be on the [column_lookups.py](https://github.com/moj-analytical-services/splink/blob/master/splink/settings_validation/column_lookups.py) script.
+
+There are currently three classes employed to construct the current log strings. These can be extended to perform additional column checks.
+
+??? note "`InvalidCols`"
+    `InvalidCols` is a NamedTuple, used to construct the bulk of our log strings. This accepts a list of columns and the type of error, producing a complete log string when requested.
+
+    In practice, this is used as follows:
+    ```py
+    # Store the invalid columns and why they're invalid
+    my_invalid_cols = InvalidCols("invalid_cols", ["first_col", "second_col"])
+    # Construct the corresponding log string
+    my_invalid_cols.construct_log_string()
+    ```
+
+??? note "`InvalidColValidator`"
+    `InvalidColValidator` houses a series of validation checks to evaluate whether the column(s) contained within either a SQL string or a user's raw input string, are present within the underlying dataframes.
+
+    To achieve this, it employs a range of cleaning functions to standardise our column inputs and conducts a series of checks on these cleaned columns. It utilises `InvalidCols` tuples to log any identified invalid columns.
+
+    It inherits from our the `SettingsValidator` class.
+
+??? note "`InvalidColumnsLogger`"
+    The principal logging class for our invalid column checks.
+
+    This class primarily calls our builder functions outlined in `InvalidColValidator`, constructing a series of log strings for output to both the console and the user's log file (if it exists).
+
+To extend the column checks, you simply need to add an additional validation method to the `InvalidColValidator` class. The implementation of this will vary depending on whether you merely need to assess the validity of a single column or columns within a SQL statement.
+
+#### Single Column Checks
+To assess single columns, the `validate_settings_column` should typically be used. This takes in a `setting_id` (analogous to the title you want to give your log string) and a list of columns to be checked.
```

Comment:
Should `valid_settings_column` have a link? Also, it is not clear what it is. A function? A class?

Reply:
Also, here I think it may be helpful to state the context at the start i.e. Checks are applied on all columns by default, but there is functionality available to apply more specific checks

Reply:
Is it helpful here to refer to as "specific column checks" or "named column checks" as this mentions being able to refer to multiple columns, not just one? So I'm not sure if "single column checks" is the clearest naming. However I may be misunderstanding!